tools/cmake/mbed_set_post_build.cmake

```diff
@@ -112,25 +91,16 @@ function(mbed_set_post_build target)
     mbed_validate_application_profile(${target})
     mbed_generate_bin_hex(${target})
 
+    if(COMMAND mbed_post_build_function)
+        # Remove the .elf file to force regenerate the application binaries
+        # (including .bin and .hex). This ensures that the post-build script runs
+        # on a raw application instead of a previous build that already went
+        # through the post-build process once.
```

Comment:
Preexisting: Seems strange that we need to do this. If the build system decides there's "nothing to do" is the post build hook going to run regardless?